jaeger-core/src/main/java/io/jaegertracing/internal/propagation/TextMapCodec.java

```diff
@@ -125,24 +125,34 @@ public JaegerSpanContext extract(TextMap carrier) {
           baggage = new HashMap<String, String>();
         }
         baggage.put(keys.unprefixedKey(key, baggagePrefix), decodedValue(entry.getValue()));
+      } else if (key.equals(Constants.BAGGAGE_HEADER_KEY)) {
+        baggage = parseBaggageHeader(decodedValue(entry.getValue()), baggage);
       }
     }
-    if (context == null) {
-      if (debugId != null) {
-        return objectFactory.createSpanContext(
-            0,
-            0,
-            0,
-            (byte) 0,
-            Collections.<String, String>emptyMap(),
-            debugId);
-      }
-      return null;
-    }
-    if (baggage == null) {
+    if (debugId == null && baggage == null) {
       return context;
     }
-    return context.withBaggage(baggage);
+    return objectFactory.createSpanContext(
+      context == null ? 0 : context.getTraceId(),
+      context == null ? 0 : context.getSpanId(),
+      context == null ? 0 : context.getParentId(),
+      context == null ? (byte)0 : context.getFlags(),
+      baggage,
+      debugId);
+  }
+
+  private Map<String, String> parseBaggageHeader(String header, Map<String, String> baggage) {
+    String[] parts = header.split("\\ *,\\ *");
+    for (int i = 0; i < parts.length; i++) {
```

Comment:
minor nit: can be simplified to a range for loop because we don't use the outer index

`for (String part : header.split(" *, *")) {`

Also, are the regex escapes required here?

Reply:
I means `\s`, will change.